src/components/Interpretations/InterpretationsProvider/InterpretationsManager.js

```diff
@@ -0,0 +1,307 @@
+                type: 'delete',
+            },
+            { onComplete, onError }
+        )
+        // This happens when deleting the interpretation from the modal
+        if (this.activeInterpretationId && id === this.activeInterpretationId) {
+            this.clearActiveInterpretation()
+        }
+        this.interpretations.delete(id)
+        this.notifyInterpretationsListObserver()
+        return null
+    }
+
+    async updateInterpretationText({ id, text, onComplete, onError }) {
+        await this.mutate(
+            {
+                resource: 'interpretations',
+                type: 'update',
+                partial: false,
+                id,
+                data: text,
+            },
+            { onComplete, onError }
+        )
+        const updatedInterpretation = {
+            ...this.getInterpretation(id),
+            text,
+        }
+        this.interpretations.set(id, updatedInterpretation)
+        this.notifyInterpretationObservers(id)
+        return updatedInterpretation
+    }
+
+    async toggleInterpretationLike(id) {
+        const interpretation = this.getInterpretation(id)
+        const wasLikedByCurrentUser = interpretation.likedBy.some(
+            (likedBy) => likedBy.id === this.currentUser.id
+        )
+        await this.mutate({
+            resource: `interpretations/${id}/like`,
+            type: wasLikedByCurrentUser ? 'delete' : 'create',
+        })
+        const isLikedByCurrentUser = !wasLikedByCurrentUser
+        const updatedInterpretation = {
+            ...interpretation,
+            likedBy: isLikedByCurrentUser
+                ? interpretation.likedBy.concat({
+                      id: this.currentUser.id,
+                  })
+                : interpretation.likedBy.filter(
+                      (lb) => lb.id !== this.currentUser.id
+                  ),
+            likes: isLikedByCurrentUser
+                ? interpretation.likes + 1
+                : interpretation.likes - 1,
+        }
+        this.interpretations.set(id, updatedInterpretation)
+        this.notifyInterpretationObservers(id)
+        return updatedInterpretation
+    }
+
+    async addCommentToActiveInterpretation({ text, onComplete }) {
+        const { id } = this.getActiveInterpretation()
+        await this.mutate(
+            {
+                resource: `interpretations/${id}/comments`,
+                type: 'create',
+                data: text,
+            },
+            { onComplete }
+        )
+        const interpretation = await this.fetchInterpretationDetails(id)
+        this.interpretations.set(id, interpretation)
+        this.notifyInterpretationObservers(id)
+        return interpretation
+    }
+
+    async deleteCommentFromActiveInterpretation({ id }) {
+        const activeInterpretation = this.getActiveInterpretation()
+        await this.mutate({
+            resource: 'interpretations',
+            id: `${activeInterpretation.id}/comments/${id}`,
```

Comment:
Any reason for not putting the whole string in `resource` as done above?
This `id` parameter containing a path is a bit weird.